src/ui/screens/ImportAccountScreen.tsx

```diff
@@ -50,49 +49,36 @@ function ImportAccount({navigation}: {navigation: NavigationProp<AccountsStackPa
   const [account, setAccountState] = React.useState<Account>({title: '', password: '', confirmPassword: ''});
   const setAccount = (_account: Account) => {
     setAccountState(_account);
-    setError('');
   };
 
   const [isPasswordVisible, setIsPasswordVisible] = React.useState(false);
   const {seed, setSeed, address, isSeedValid} = useParseSeed();
   const {addAccount} = useAccounts();
-  const [error, setError] = React.useState<null | string>(null);
 
   const passwordStrength = zxcvbn(account.password).score;
 
-  const onSubmit = async () => {
-    if (!isSeedValid) {
-      setError('The seed appears to be invalid');
-    } else if (!address) {
-      setError('Could not parse the address from the seed');
-    } else if (!account.password) {
-      setError('Please enter a password');
-    } else if (account.password !== account.confirmPassword) {
-      setError('Passwords do not match');
-    } else if (passwordStrength < 3) {
-      setError('Password is too weak');
-    } else {
-      if (!address) {
-        throw new Error('address not provided');
-      }
-      const encoded = await SubstrateSign.encryptData(seed, account.password);
-      const _address = await SubstrateSign.substrateAddress(seed, currentNetwork.ss58Format);
+  const seedError = Boolean(seed) && !isSeedValid;
+  const passwordError = Boolean(account.password) && passwordStrength < 3;
+  const confirmPasswordError = Boolean(account.confirmPassword) && !(account.password === account.confirmPassword);
 
-      const newAcc = {
-        address: _address,
-        encoded,
-        meta: {
-          name: account.title,
-          network: currentNetwork.key,
-          isFavorite: false,
-        },
-        isExternal: false,
-      };
+  const isDisabled = !isSeedValid || !account.password || !(account.password === account.confirmPassword);
 
-      addAccount(newAcc);
-      SecureKeychain.setPasswordByServiceId(account.password, 'BIOMETRICS', address);
-      navigation.navigate(accountsScreen, {reload: true});
-    }
+  const onSubmit = async () => {
+    const encoded = await SubstrateSign.encryptData(seed, account.password);
+    const _address = await SubstrateSign.substrateAddress(seed, currentNetwork.ss58Format);
+    const newAcc = {
+      address: _address,
+      encoded,
+      meta: {
+        name: account.title,
+        network: currentNetwork.key,
+        isFavorite: false,
+      },
+      isExternal: false,
+    };
+    addAccount(newAcc);
+    SecureKeychain.setPasswordByServiceId(account.password, 'BIOMETRICS', _address);
```

Comment:
@franjs 
I thought we were using the formatted address in the secure key chain for import account, no?